scope_test.go

```diff
@@ -529,6 +575,36 @@ func TestTaggedSubScope(t *testing.T) {
 	}, r.histograms["foo.bar"].tags)
 }
 
+func TestTaggedSanitisedSubScope(t *testing.T) {
+	r := newTestStatsReporter()
+
+	ts := map[string]string{"env": "test:env"}
+	root, closer := NewRootScope(ScopeOptions{
+		Prefix:          "foo",
+		Tags:            ts,
+		Reporter:        r,
+		SanitiseOptions: &alphanumericSanitiserOpts,
+	}, 0)
+	defer closer.Close()
+
+	s := root.(*scope)
+
+	tscope := root.Tagged(map[string]string{"service": "test.service"}).(*scope)
+
+	r.cg.Add(1)
+	tscope.Counter("beep").Inc(1)
+
+	s.report(r)
+	tscope.report(r)
+	r.cg.Wait()
+
+	assert.EqualValues(t, 1, r.counters["foo.beep"].val)
```

Comment:
I'll buy that. Behaviour will be easier to explain if we sanitise everything too. Updated the code to do the sanitisation post `fullyQualifiedName`